core/src/main/scala/laserdisc/protocol/ServerP.scala

```diff
@@ -118,34 +130,37 @@ object ServerP {
 
   final case class Info(sections: Map[InfoSection, Parameters])
   final object Info {
-    private val ISR: String ==> InfoSection = Read.instancePF {
-      case "server"       => InfoSection.server
-      case "clients"      => InfoSection.clients
-      case "memory"       => InfoSection.memory
-      case "persistence"  => InfoSection.persistence
-      case "stats"        => InfoSection.stats
-      case "replication"  => InfoSection.replication
-      case "cpu"          => InfoSection.cpu
-      case "commandstats" => InfoSection.commandstats
-      case "cluster"      => InfoSection.cluster
-      case "keyspace"     => InfoSection.keyspace
+    private val ISR: String ==> InfoSection = Read.instance {
+      case "server"       => Right(InfoSection.server)
+      case "clients"      => Right(InfoSection.clients)
+      case "memory"       => Right(InfoSection.memory)
+      case "persistence"  => Right(InfoSection.persistence)
+      case "stats"        => Right(InfoSection.stats)
+      case "replication"  => Right(InfoSection.replication)
+      case "cpu"          => Right(InfoSection.cpu)
+      case "commandstats" => Right(InfoSection.commandstats)
+      case "cluster"      => Right(InfoSection.cluster)
+      case "keyspace"     => Right(InfoSection.keyspace)
+      case other          => Left(Err(s"Unexpected server info specification. Was $other"))
     }
     private val KVPair = "(.*):(.*)".r
-    private val PR: Seq[String] ==> Parameters = Read.instancePF {
-      case ss => new Parameters(ss.collect { case KVPair(k, v) => k -> v }.toMap)
+    private val PR: Seq[String] ==> Parameters = Read.instance { ss =>
+      Right(new Parameters(ss.collect { case KVPair(k, v) => k -> v }.toMap))
     }
 
-    implicit val infoRead: Bulk ==> Info = Read.instancePF {
+    implicit val infoRead: Bulk ==> Info = Read.instance {
       case Bulk(s) =>
-        Info(
-          s.split(LF * 2)
-            .flatMap {
-              _.split(LF_CH).toSeq match {
-                case ISR(infoSection) +: PR(parameters) => Some(infoSection -> parameters)
-                case _                                  => None
+        Right(
+          Info(
+            s.split(LF * 2)
+              .flatMap {
+                _.split(LF_CH).toSeq match {
+                  case ISR(Right(infoSection)) +: PR(Right(parameters)) => Some(infoSection -> parameters)
```

Comment:
^^ again, you're ignoring the `Left`s, should we maybe consider the above `private val x: A ==> B` (i.e. the `ISR` and `PR`) as `PartialFunction[A, B]` in case we can't do better than ignoring?

Reply:
this is wrong. I will fix it.